app/menus/members/menu.rb

```diff
@@ -24,50 +24,49 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
 # See COPYRIGHT and LICENSE files for more details.
-#++
+# ++
+module Members
+  class Menu < Submenu
+    attr_reader :project, :params
 
-module Menus
-  module MembersHelper
-    def first_level_menu_items
-      [OpenProject::Menu::MenuGroup.new(header: nil, children: user_status_options)] + nested_menu_items
-    end
-
-    private
+    def initialize(project: nil, params: nil)
+      @project = project
+      @params = params
 
-    def user_status_options
-      [
-        OpenProject::Menu::MenuItem.new(title: I18n.t("members.menu.all"),
-                                        href: project_members_path,
-                                        selected: active_filter_count == 0),
-        OpenProject::Menu::MenuItem.new(title: I18n.t("members.menu.locked"),
-                                        href: project_members_path(status: :locked),
-                                        selected: selected?(:status, :locked)),
-        OpenProject::Menu::MenuItem.new(title: I18n.t("members.menu.invited"),
-                                        href: project_members_path(status: :invited),
-                                        selected: selected?(:status, :invited))
-      ]
+      super(view_type:, project:, params:)
     end
 
-    def nested_menu_items
+    def menu_items
       [
+        OpenProject::Menu::MenuGroup.new(header:nil, children: user_status_options),
         OpenProject::Menu::MenuGroup.new(header: I18n.t("members.menu.project_roles"), children: project_roles_entries),
         OpenProject::Menu::MenuGroup.new(header: I18n.t("members.menu.wp_shares"), children: permission_menu_entries),
         OpenProject::Menu::MenuGroup.new(header: I18n.t("members.menu.groups"), children: project_group_entries)
       ]
     end
 
+    def user_status_options
+      [
+        OpenProject::Menu::MenuItem.new(title: I18n.t("members.menu.all"),
+                                        href: project_members_path(project),
+                                        selected: active_filter_count == 0),
+        menu_item({status: :locked}, I18n.t("members.menu.locked")),
+        menu_item({status: :invited}, I18n.t("members.menu.invited"))
+      ]
+    end
+
     def project_roles_entries
       ProjectRole
         .where(id: MemberRole.where(member_id: @project.members.select(:id)).select(:role_id))
         .distinct
         .pluck(:id, :name)
-        .map { |id, name| menu_item(:role_id, id, name) }
+        .map { |id, name| menu_item( { role_id: id }, name) }
```

Comment:
Probably minor, but I do not like the order of the parameters of this `menu_item` method. I feel like the name should some first, and then the parameters. It would also match the `OpenProject::Menu::MenuItem` constructor where `title` comes first.

It would become `menu_item(name, role_id: id)` (instead of `menu_item( { role_id: id }, name)`)

I can add a commit to refactor it if it can help.

Reply:
Done 👍